pytket/extensions/cutensornet/tnstate/ttn.py

```diff
@@ -0,0 +1,854 @@
+
+        for i, q in enumerate(ilo_qubits):
+            # Create the tensors for each qubit in ``state``
+            bitvalue = 1 if state & 2 ** (len(ilo_qubits) - i - 1) else 0
+            tensor = cp.zeros(shape=(2,), dtype=self._cfg._complex_t)
+            tensor[bitvalue] = 1
+            # Append it to the interleaved representation
+            interleaved_rep.append(tensor)
+            interleaved_rep.append([str(q)])  # The bond
+        # Ignore the dim=1 tensors in the output
+        interleaved_rep.append([])
+
+        # Contract
+        result = cq.contract(
+            *interleaved_rep,
+            options={"handle": self._lib.handle, "device_id": self._lib.device_id},
+            optimize={"samples": 1},  # There is little to no optimisation to be done
+        )
+
+        self._logger.debug(f"Amplitude of state {state} is {result}.")
+        return complex(result)
+
+    def get_qubits(self) -> set[Qubit]:
+        """Returns the set of qubits that this TTN is defined on."""
+        return set(self.qubit_position.keys())
+
+    def get_interleaved_representation(
+        self, conj: bool = False
+    ) -> list[Union[Tensor, str]]:
+        """Returns the interleaved representation of the TTN used by cuQuantum.
+
+        Args:
+            conj: If True, all tensors are conjugated and bonds IDs are prefixed
+                with * (except physical bonds). Defaults to False.
+        """
+        self._logger.debug("Creating interleaved representation...")
+
+        # Auxiliar dictionary of physical bonds to qubit IDs
+        qubit_id = {
+            location: str(qubit) for qubit, location in self.qubit_position.items()
+        }
+
+        interleaved_rep = []
+        for path, node in self.nodes.items():
+            # Append the tensor
+            if conj:
+                interleaved_rep.append(node.tensor.conj())
+            else:
+                interleaved_rep.append(node.tensor)
+
+            # Create the ID for the parent bond
+            parentID = "".join(str(int(d)) for d in path)
+            if conj:
+                parentID = "*" + parentID
+
+            # Append the bonds
+            if node.is_leaf:
+                bonds = []
+                for b in range(len(node.tensor.shape) - 1):
+                    bonds.append(qubit_id[(path, b)])
+                bonds.append(parentID)
+            else:
+                bonds = [parentID + "0", parentID + "1", parentID]
+
+            interleaved_rep.append(bonds)
+            self._logger.debug(f"Bond IDs: {bonds}")
+
+        return interleaved_rep
+
+    def get_dimension(self, path: RootPath, direction: DirTTN) -> int:
+        """Returns the dimension of bond ``dir`` of the node at ``path``.
+
+        Args:
+            path: The path to a node in the TTN.
+            direction: The direction of the bond.
+
+        Returns:
+            The dimension of the bond between the node and its parent.

```

Comment:
Isn't that contradicting the title (as in, not necessarily of the bond between the node and parent)?

Reply:
Oops, yes. The text in the Returns line is outdated.

Reply:
Done